test/addons/dlopen-ping-pong/test.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+const common = require('../../common');
+
+if (common.isWindows) {
+  common.skip('dlopen global symbol loading is not supported on this os.');
+  return;
+}
+
+const assert = require('assert');
+const path = require('path');
+const os = require('os');
+
+const bindingPath = require.resolve(`./build/${common.buildType}/binding`);
+process.dlopen(module, bindingPath,
+               os.constants.dlopen.RTLD_NOW | os.constants.dlopen.RTLD_GLOBAL);
+module.exports.load(path.dirname(bindingPath) + '/ping.so');
+assert.strictEqual(module.exports.ping(), 'pong');
+console.log('binding.ping() =', module.exports.ping());
```

Comment:
@jasnell You sure that would belong on this test? The result will be an exception with error `Error: Module did not self-register.`.
If needed, I'd say it goes in its own test, but to be honest I'm not entirely sure what's the value of it.